crypto/evp_extra/evp_asn1.c

```diff
@@ -274,6 +274,48 @@ static EVP_PKEY *old_priv_decode(CBS *cbs, int type) {
   return NULL;
 }
 
+int EVP_PKEY_check(EVP_PKEY_CTX *ctx) {
+  EVP_PKEY *pkey = ctx->pkey;
+
+  if (pkey == NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_NO_KEY_SET);
+    return 0;
+  }
+
+  switch(pkey->type) {
+    case EVP_PKEY_EC:
+      if (EC_KEY_get0_private_key(pkey->pkey.ec) == NULL) {
```

Comment:
why do we need this?

Reply:
EVP_PKEY_check is supposed to validate both the private and public aspects of the key as opposed to EVP_PKEY_public_check which only validates the public component of the key pair. EC_KEY_check_key [doesn't fail ](https://github.com/aws/aws-lc/blob/13337f78937bc5911905e71d58ce9c71c23166ee/crypto/fipsmodule/ec/ec_key.c#L310)when priv_key is null, so there is an additional check before.

Reply:
RSA_check_key also doesn't fail when given a public key only.

Reply:
> EVP_PKEY_check is supposed to validate both the private and public aspects

so OpenSSL fails when `EVP_PKEY_check` is called with public key only?